### ISSUE TITLE
Web - Order of task notification messages incorrect when back online

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21314]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any report
2. Create a task
3. Setting => Preferences =>  Force offline
4. Click Mark as done
5. Click three dot => mark as incomplete
6. Click  three dot => Cancel
7. Setting => Preferences =>  Toggle Force Offline => Close
8. Observe the message order:completed task => reopened task => canceled task
9. Reload page: Observe the message order: cancel task => task completed => task reopened

## Expected Result:
After reload page the message order is correct: completed task => reopened task => canceled task

## Actual Result:
After reload page message order is not correct

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.28-4

**Reproducible in staging?:** n/a

**Reproducible in production?:** n/a

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/413bf848-870b-4b1b-9a8a-7a6f27d39eb5

**Expensify/Expensify Issue URL:**

**Issue reported by:** @namhihi237

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686889606135589

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lschurr** [comment]
I was able to reproduce:

<img width="306" alt="image" src="https://github.com/Expensify/App/assets/33293730/b8035d57-186c-463b-b170-4f7d54c47888">


<img width="342" alt="image" src="https://github.com/Expensify/App/assets/33293730/8ecccd17-3ef7-4bed-9cdd-f5517d466637">

**Comment by lschurr** [comment]
Could you review this proposal @rushatgabhane?

**Comment by thienlnam** [comment]
This is not being worked on as part of the refactor - great catch that it's not using the provided optimistic reportActionID though!

**Comment by lschurr** [comment]
Should this be closed @thienlnam?

**Comment by thienlnam** [comment]
Nope, there are a couple pieces to this that will require internal changes as well as possible external changes

**Comment by hayata-suenaga** [comment]
Let's make this internal and if App change is required, we can create another issue for that.

**Comment by hayata-suenaga** [comment]
Asked help from [the internal team here](https://expensify.slack.com/archives/C03TQ48KC/p1688175494873119)

**Comment by lschurr** [comment]
Still in progress @hayata-suenaga?

**Comment by thienlnam** [comment]
Taking this over from Hayata

**Comment by thienlnam** [comment]
I've tested this locally a couple things. It seems like complete and reopen task is using the optimistic reportAction correctly. 

We're updating the cancel task flow [here](https://github.com/Expensify/App/issues/19504) so we don't need to handle that here. Does the problem just lie with the Cancelled task action?

**Comment by lschurr** [comment]
Any update here @thienlnam?

**Comment by thienlnam** [comment]
This is an internal issue, but also should hold for https://github.com/Expensify/App/issues/19504 where we update the 'cancel' task

**Comment by lschurr** [comment]
Cool, so this is on hold right @thienlnam?

**Comment by thienlnam** [comment]
Actually, I think this should be fixed in https://github.com/Expensify/App/pull/20841 so we can wait for that to get to production and then can re-test this

**Comment by thienlnam** [comment]
Cool, thanks for testing - we can go with your proposal to fix that last part!

**Comment by thienlnam** [comment]
@namhihi237 DM me the email you used to sign up and I can work on figuring out why

**Comment by lschurr** [comment]
Payout summary:

- Issue reporter: @namhihi237 $250
- Contributor: @hoangzinh $1000
- C+: @rushatgabhane $1000

@namhihi237 and @hoangzinh will be paid in Upwork: https://www.upwork.com/jobs/~017368db113f47cd42
@rushatgabhane please request payment via ND if you haven't already

**Comment by lschurr** [comment]
Also @rushatgabhane could you work on the checklist? https://github.com/Expensify/App/issues/21314#issuecomment-1642507060

**Comment by lschurr** [comment]
Paid @namhihi237 and @hoangzinh

We can close once you are able to work through the checklist @rushatgabhane

**Comment by lschurr** [comment]
Bump @rushatgabhane

**Comment by rushatgabhane** [comment]
WIP ⚠️ 

1. The PR that introduced the bug has been identified. Link to the PR:

2. The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

3. A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

4. Determine if we should create a regression test for this bug.

5. If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again

**Comment by rushatgabhane** [comment]
created a manual request https://staging.new.expensify.com/r/6056438555742596

**Comment by JmillsExpensify** [comment]
Reviewed details for @rushatgabhane. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.
